### PR TITLE
Publish improvements

### PR DESCRIPTION
**Pull request by cvogt**

### CODE REVIEW

File: stage2/License.scala [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package cbt
+import java.net.URL
+case class License(name: String, url: URL)
+object License{
+  object Apache2 extends License(
+    "Apache-2.0",
+    new URL(s"http://www.apache.org/licenses/LICENSE-2.0")
+  )
```
Comment (cvogt) [inline stage2/License.scala]: @MasseGuillaume check this out. This addresses https://github.com/cvogt/cbt/issues/67. Now we only need to add all the licenses with nice identifiers. Would you like to contribute that? Any ideas for the exact identifiers we choose? I find the nix ones too cryptic. Is there any standard we can use or just invent our own?

Comment (MasseGuillaume) [inline stage2/License.scala]: Nix uses http://spdx.org/licenses/ that's one standard.

```scala
case class License(name: String, shortName: String, url: Option[String])
def spdx(id: String, name: String) = License(name, id, Some(s"https://spdx.org/licenses/$id.html"))
val Academic = spdx("AFL-3.0", "Academic Free License")
val Affero = spdx("AGPL-3.0", "GNU Affero General Public License v3.0")
val Apache2 = spdx("Apache-2.0", "Apache License 2.0")
val Apple2_0 = spdx("APSL-2.0", "Apple Public Source License 2.0")
val Beerware = spdx("Beerware", "Beerware License")
val Bsd2Clause = spdx("BSD-2-Clause", """BSD 2-clause "Simplified" License""")
val Bsd3Clause = spdx("BSD-3-Clause", """BSD 3-clause "New" or "Revised" License""")
val BsdOriginal = spdx("BSD-4-Clause", """BSD 4-clause "Original" or "Old" License""")
val CreativeCommonsZeroUniversal = spdx("CC0-1.0", "Creative Commons Zero v1.0 Universal")
val CreativeCommonsAttributionNonCommercialShareAlike_2_0 = spdx("CC-BY-NC-SA-2.0", "Creative Commons Attribution Non Commercial Share Alike 2.0")
val CreativeCommonsAttributionNonCommercialShareAlike_2_5 = spdx("CC-BY-NC-SA-2.5", "Creative Commons Attribution Non Commercial Share Alike 2.5")
val CreativeCommonsAttributionNonCommercialShareAlike_3_0 = spdx("CC-BY-NC-SA-3.0", "Creative Commons Attribution Non Commercial Share Alike 3.0")
val CreativeCommonsAttributionNonCommercialShareAlike_4_0 = spdx("CC-BY-NC-SA-4.0", "Creative Commons Attribution Non Commercial Share Alike 4.0")
val CreativeCommonsAttributionShareAlike_2_5 = spdx("CC-BY-SA-2.5", "Creative Commons Attribution Share Alike 2.5")
val CreativeCommonsAttribution_3_0 = spdx("CC-BY-3.0", "Creative Commons Attribution 3.0")
val CreativeCommonsAttributionShareAlike_3_0 = spdx("CC-BY-SA-3.0", "Creative Commons Attribution Share Alike 3.0")
val CreativeCommonsAttribution_4_0 = spdx("CC-BY-4.0", "Creative Commons Attribution 4.0")
val CreativeCommonsAttributionShareAlike_4_0 = spdx("CC-BY-SA-4.0", "Creative Commons Attribution Share Alike 4.0")
val Eclipse = spdx("EPL-1.0", "Eclipse Public License 1.0")
val GPL1 = spdx("GPL-1.0", "GNU General Public License v1.0 only")
val GPL1Plus = spdx("GPL-1.0+", "GNU General Public License v1.0 or later")
val GPL2 = spdx("GPL-2.0", "GNU General Public License v2.0 only")
val GPL2Plus = spdx("GPL-2.0+", "GNU General Public License v2.0 or later")
val GPl3 = spdx("GPL-3.0", "GNU General Public License v3.0 only")
val GPL3Plus = spdx("GPL-3.0+", "GNU General Public License v3.0 or later")
val ISC = spdx("ISC", "ISC License")
val LGPL2 = spdx("LGPL-2.0", "GNU Library General Public License v2 only")
// @deprecated("-", "-")
val LGPL2_Plus = spdx("LGPL-2.0+", "GNU Library General Public License v2 or later")
val LGPL2_1 = spdx("LGPL-2.1", "GNU Library General Public License v2.1 only")
// @deprecated("-", "-")
val LGPL2_1_Plus = spdx("LGPL-2.1+", "GNU Library General Public License v2.1 or later")
val LGPL3 = spdx("LGPL-3.0", "GNU Lesser General Public License v3.0 only")
// @deprecated("use LGPL3", "2.0rc2")
val LGPL3_Plus = spdx("LGPL-3.0+", "GNU Lesser General Public License v3.0 or later")
// Spdx.org does not (yet) differentiate between the X11 and Expat versions
// for details see http://en.wikipedia.org/wiki/MIT_License#Various_versions
val MIT = spdx("MIT", "MIT License")
val MPL_1_0 = spdx("MPL-1.0", "Mozilla Public License 1.0")
val MPL_1_1 = spdx("MPL-1.1", "Mozilla Public License 1.1")
val MPL2 = spdx("MPL-2.0", "Mozilla Public License 2.0")
val PublicDomain = License("Public Domain", "Public Domain", None)
val Scala = License ("Scala License", "Scala License", Some("http://www.scala-lang.org/license.html"))
val TypesafeSubscriptionAgreement = License("Typesafe Subscription Agreement", "Typesafe Subscription Agreement", Some("http://downloads.typesafe.com/website/legal/TypesafeSubscriptionAgreement.pdf"))
val Unlicense = spdx("Unlicense", "The Unlicense")
val W3C = spdx("W3C", "W3C Software Notice and License")
val WTFPL = spdx("WTFPL", "Do What The F*ck You Want To Public License")
```

File: stage2/PublishBuild.scala [hunk 2]
```diff
@@ -5,13 +5,15 @@ import java.nio.file.Files.readAllBytes
...
   def description: String
   def url: URL
   def developers: Seq[Developer]
   def licenses: Seq[License]
   def scmUrl: String
   def scmConnection: String
+  def inceptionYear: Int
+  def organizationName: String
+  def organizationUrl: URL
   def pomExtra: Seq[scala.xml.Node] = Seq()
```
Comment (MasseGuillaume) [inline stage2/PublishBuild.scala]: I would kill this one from the start.

File: stage2/Lib.scala [hunk 3]
```diff
@@ -307,6 +306,9 @@ final class Lib(logger: Logger) extends Stage1Lib(logger) with Scaffold{
     licenses: Seq[License],
     scmUrl: String, // seems like invalid URLs are used here in pom files
     scmConnection: String,
```
Comment (MasseGuillaume) [inline stage2/Lib.scala]:
```scala
case class SourceCodeManagment(
  connection: Option[String],
  developerConnection: Option[String],
  url: Option[String],
  tag: Option[String]
)
```

File: stage2/Lib.scala [hunk 4]
```diff
@@ -344,6 +346,11 @@ final class Lib(logger: Logger) extends Stage1Lib(logger) with Scaffold{
             <url>{scmUrl}</url>
             <connection>{scmConnection}</connection>
           </scm>
+          <inceptionYear>{inceptionYear}</inceptionYear>
+          <organization>
```
Comment (MasseGuillaume) [inline stage2/Lib.scala]: this one is optionnal

File: stage2/PublishBuild.scala [hunk 5]
```diff
@@ -5,13 +5,15 @@ import java.nio.file.Files.readAllBytes
...
 abstract class PublishBuild(context: Context) extends PackageBuild(context){
-  def name = artifactId
   def description: String
   def url: URL
   def developers: Seq[Developer]
   def licenses: Seq[License]
   def scmUrl: String
   def scmConnection: String
+  def inceptionYear: Int
+  def organizationName: String
```
Comment (MasseGuillaume) [inline stage2/PublishBuild.scala]:
```scala
organization: Option[Organization]

case class Organization(
  name: String,
  url: Option[String]
)
```

File: stage2/Lib.scala (hunk 3 above)
Comment (cvogt) [inline stage2/Lib.scala]: where does tag come from? is that spec'ed somewhere?

File: stage2/PublishBuild.scala (hunk 2 above)
Comment (cvogt) [inline stage2/PublishBuild.scala]: done

File: stage2/Lib.scala (hunk 4 above)
Comment (cvogt) [inline stage2/Lib.scala]: done

File: stage2/PublishBuild.scala (hunk 5 above)
Comment (cvogt) [inline stage2/PublishBuild.scala]: done

File: stage2/License.scala [hunk 6]
```diff
@@ -1,9 +1,57 @@
...
+  val LGPL3_Plus = spdx("LGPL-3.0+", "GNU Lesser General Public License v3.0 or later")
+  /** Spdx.org does not (yet) differentiate between the X11 and Expat versions
+      for details see http://en.wikipedia.org/wiki/MIT_License#Various_versions */
+  val MIT = spdx("MIT", "MIT License")
+  val MPL_1_0 = spdx("MPL-1.0", "Mozilla Public License 1.0")
+  val MPL_1_1 = spdx("MPL-1.1", "Mozilla Public License 1.1")
+  val MPL2 = spdx("MPL-2.0", "Mozilla Public License 2.0")
+  val Unlicense = spdx("Unlicense", "The Unlicense")
+  val W3C = spdx("W3C", "W3C Software Notice and License")
+  val WTFPL = spdx("WTFPL", "Do What The F*ck You Want To Public License")
```
Comment (cvogt) [inline stage2/License.scala]: @MasseGuillaume the names seem partly a bit irregular: Apache2, Apple2_0, MPL_1_0. Where did you get these names from? Should we unify them?

File: stage2/Lib.scala (hunk 3 above)
Comment (cvogt) [inline stage2/Lib.scala]: waiting with this change until more info accumulates.

Comment (MasseGuillaume) [inline stage2/Lib.scala]: http://maven.apache.org/components/ref/3.3.9/maven-model/maven.html#scm

```
tag	String	The tag of current code. By default, it's set to HEAD during development.
Default value is: HEAD.
```

File: stage2/License.scala (hunk 6 above)
Comment (MasseGuillaume) [inline stage2/License.scala]: the identifier on the left I made them up.

they could map more to the spdx identitfier
`Apache-2.0` => `Apache_2_0`
maybe with some alias
`val Apache = Apache_2_0`

File: stage2/Lib.scala (hunk 3 above)
Comment (cvogt) [inline stage2/Lib.scala]: How far would we take this though? Which parts of the maven spec to include? Is tag something any Scala library publishes?

**Pull request merged**